lib/presentation/pages/tip_resources_page.dart

```diff
@@ -31,9 +34,17 @@ class TipResourcePage extends StatelessWidget {
       margin: false,
       topBarType: TopBarType.back,
       expendChild: false,
+      topBarRightIcon: Icons.open_in_new,
+      topBarRightOnTap: () => openUrl(url),
       child: WebViewMolecule(
         url: url,
       ),
     );
   }
+
+  Future<void> openUrl(String url) async {
+    if (!await launchUrl(Uri.parse(url))) {
+      throw Exception('Could not launch $url');
+    }
+  }
```

Comment:
Can you create a controller that calls launchUrl methods?.
Follow the existing controllers as an example.